### ISSUE TITLE
PostgreSQL - QueryFailedError: relation "pg_matviews" does not exist

### ISSUE THREAD

**Issue by H6LS1S** [typeorm/typeorm#7714]
## Issue Description
After updating from 0.2.32 to [0.2.33](https://github.com/typeorm/typeorm/compare/0.2.33...master)-[34](https://github.com/typeorm/typeorm/compare/0.2.34...master), the pg driver tries to select a non-existent relation "pg_matviews"

### Actual Behavior
```sh
query: START TRANSACTION
query: SELECT 'DROP VIEW IF EXISTS "' || schemaname || '"."' || viewname || '" CASCADE;' as "query" FROM "pg_views" WHERE "schemaname" IN (current_schema()) AND "viewname" NOT IN ('geography_columns', 'geometry_columns', 'raster_columns', 'raster_overviews')
query: SELECT 'DROP MATERIALIZED VIEW IF EXISTS "' || schemaname || '"."' || matviewname || '" CASCADE;' as "query" FROM "pg_matviews" WHERE "schemaname" IN (current_schema())
query failed: SELECT 'DROP MATERIALIZED VIEW IF EXISTS "' || schemaname || '"."' || matviewname || '" CASCADE;' as "query" FROM "pg_matviews" WHERE "schemaname" IN (current_schema())
error: error: relation "pg_matviews" does not exist
    at Parser.parseErrorMessage (/home/node_modules/pg-protocol/dist/parser.js:287:98)
    at Parser.handlePacket (/home/node_modules/pg-protocol/dist/parser.js:126:29)
    at Parser.parse (/home/node_modules/pg-protocol/dist/parser.js:39:38)
    at Socket.<anonymous> (/home/node_modules/pg-protocol/dist/index.js:11:42)
    at Socket.emit (events.js:376:20)
    at addChunk (internal/streams/readable.js:309:12)
    at readableAddChunk (internal/streams/readable.js:284:9)
    at Socket.Readable.push (internal/streams/readable.js:223:10)
    at TCP.onStreamRead (internal/stream_base_commons.js:188:23) {
  length: 103,
  severity: 'ERROR',
  code: '42P01',
  detail: undefined,
  hint: undefined,
  position: '115',
  internalPosition: undefined,
  internalQuery: undefined,
  where: undefined,
  schema: undefined,
  table: undefined,
  column: undefined,
  dataType: undefined,
  constraint: undefined,
  file: 'parse_relation.c',
  line: '874',
  routine: 'parserOpenTable'
}
query: ROLLBACK
```

### Steps to Reproduce
1. Run `npm i typeorm@0.2.33`
2. Create docker-compose.yaml
```yaml
version: '3.5'
services:
  postgres:
    image: mdillon/postgis:9.2-alpine
    ports:
      - '5432:5432'
    environment:
      POSTGRES_DB: database
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: postgres
```
3. Use this TypeORM options:
```json
{
  "name": "development",
  "type": "postgres",
  "host": "localhost",
  "port": 5432,
  "cache": true,
  "logging": "all",
  "database": "database",
  "username": "postgres",
  "password": "postgres",
  "dropSchema": true,
  "synchronize": true,
  "migrationsRun": true,
}
``` 

### My Environment
| Dependency          | Version  |
| ---                 | ---      |
| Operating System    |     Solus OS 4.2     |
| Node.js version     | 14.17.0 |
| Typescript version  | 4.3.2 |
| TypeORM version     | 0.2.23^ |
| PostGIS | 9.2 |

### Relevant Database Driver(s)
- [x] `postgres`

### Are you willing to resolve this issue by submitting a Pull Request?
- [x] No, I don't have the time, although I believe I could do it if I had the time...

**Comment by AlexMesser** [comment]
`pg_matviews` was added in `PostgreSQL 9.3`. You need to update your Postgres docker image to a newer version.

**Comment by H6LS1S** [comment]
@AlexMesser, It seems to me that there should be backwards compatibility other  for database versions

**Comment by AlexMesser** [comment]
[Postgres 9.2](https://www.postgresql.org/support/versioning/) official support was ended in 2017. We can't keep backward compatibility for all existing database versions.

**Comment by H6LS1S** [comment]
@AlexMesser Maybe it's worth mentioning this in the documentation? It's just that many cloud providers support outdated Postgres versions, not to mention situations where it is possible to update the ORM, but there is no way to update the RDMS.

**Comment by pleerock** [comment]
@H6LS1S what action leads to this error? Calling `connection.dropDatabase()` ?

**Comment by pleerock** [comment]
should be fixed by #7716 in `0.2.35`. Please stick with `0.2.32` for now.

**Comment by jcrben** [comment]
Will databases generally be supported past EOL? Seems like a consistent policy is better here. Better to focus on supporting current
